app/controllers/shared/api.php

```diff
@@ -318,6 +318,39 @@
         }
     });
 
+App::shutdown()
+    ->groups(['session'])
```

Comment:
Can we add a description to each hook explaining its responsibility?

Reply:
I'll add to this one, but others are handling so many things

Reply:
Agree. Eventually I think we should start splitting them to have one responsibility.